airflow/models.py

```diff
@@ -1288,19 +1288,19 @@ def run(
             verbose=True)
 
         if not runnable and not mark_success:
-            if self.state != State.QUEUED:
-                # If a task's dependencies are met but it can't be run yet then queue it
-                # instead
-                self.state = State.QUEUED
-                msg = "Queuing attempt {attempt} of {total}".format(
-                    attempt=self.try_number % (task.retries + 1) + 1,
-                    total=task.retries + 1)
-                logging.info(hr + msg + hr)
-
-                self.queued_dttm = datetime.now()
-                msg = "Queuing into pool {}".format(self.pool)
-                logging.info(msg)
-                session.merge(self)
+            # we might have hit concurrency limits, which means we probably
+            # have been running prematurely.
+            self.state = State.UP_FOR_RETRY
```

Comment:
I actually like NONE or SCHEDULED better than UP_FOR_RETRY. These ones wouldn't confuse users (though ideally we would keep the QUEUED state for now and just allow the scheduler to handle it until the more long-term scheduler fix). Regarding the back-off I don't think this is needed as the scheduler should take care of this automatically (and actually should not schedule tasks beyond concurrency limits), and I'm concerned there are some edge cases with the retry state that might not be handled correctly here (e.g. I haven't looked but if retry number isn't bumped/end date isn't set not sure if some wonky behavior could happen).